### PR TITLE
FIX pickle getset_descriptor

### PR DESCRIPTION
**Pull request by pierreglaser**
preparation work for #282. Support the pickling of yet another property flavor.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -4,6 +4,9 @@
 - Fix a bug affecting bound classmethod saving on Python 2.
   ([issue #288](https://github.com/cloudpipe/cloudpickle/issues/288))
 
+- Add support for pickling exotic builtin property types.
```
Comment (ogrisel) [inline CHANGES.md]: Let's be more objective and descriptive: "add support for pickling "getset" descriptors.

**Comment by ogrisel** [comment]
By reading #291 I found this code can generate an instance of `getset_descriptor`:

```python
>>> class A: 
...     a = 1 
...                                                                                                                                                                             
>>>                                                                                                                                                                             
>>> type(A.__dict__['__dict__'])                                                                                                                                                
<class 'getset_descriptor'>
>>> type(A.__dict__)                                                                                                                                                            
<class 'mappingproxy'>
```

but this is a not a natural code usage.

**Comment by ogrisel** [comment]
I assume getset descriptors are only used by low-level builtin classes such as object and float? Is that right?

**Comment by pierreglaser** [comment]
yes. It is a builtin property type that does not even provide a constructor at the python level, so it is guaranteed to be implemented for builtin-objects only. 
Adding such a reducer allows us not to special case such properties when having to pickling them, which makes the code more readable, as I am trying to do in #291 . The reducer follows the classic builtin-property/method reducing pattern.

**Comment by pierreglaser** [comment]
> but this is a not a natural code usage.

No, but  `cloudpickle.dumps(A)`, saves `A.__dict__` is saved, including `A.__dict__['__dict__']`. Right now in cloudpickle master,  we special-case the  `__dict__` item of  `A.__dict__` and in most cases we decide not to save it, which is a bug. 
This PR does not arise directly from a user-level usage of cloudpickle - it is ground work to solve a bug.

**Comment by ogrisel** [comment]
Alright, let's merge then.

**Pull request merged**